helix-view/src/gutter.rs

```diff
@@ -56,10 +88,13 @@ pub fn line_numbers<'doc>(
     view: &View,
     theme: &Theme,
     is_focused: bool,
-    width: usize,
 ) -> GutterFn<'doc> {
+    const ELLIPSIS: char = '\u{2026}';
+
     let text = doc.text().slice(..);
     let last_line = view.last_line(doc);
+    let width = line_numbers_width(view);
```

Comment:
Wouldn't it be sufficient to call this with `text.len_lines()`, without having to refactor?

Reply:
Sure, that works. I was basing it off of the first suggestion here ([2438](https://github.com/helix-editor/helix/issues/2583#issuecomment-1139625428)), but I can do the latter suggestion instead. 

> make the width dynamic based on the line numbers within view or the maximum line number of the buffer

Reply:
I'm having trouble making this fix. 

It's important that the width here is the same as how it's calculated for `GutterType.width` used to calculate `gutter_offset`. If these provide different widths, the gutter and numbers don't stay aligned.

But I couldn't find an elegant way to provide the `Document` to `GutterType::LineNumber.width` in a way that doesn't require some major changes. But I'm still quite new to the code base, so perhaps there's some obvious solution I'm missing.

Reply:
Revisited this and it ended up being less problematic than I first thought. I needed to pass the `Document` to `View.inner_area`. This seemed fine in all but one place, where I think the total `View` area is a fine substitution. I'll highlight it in a separate thread.